task/net/src/bsp/sidecar_1.rs

```diff
@@ -0,0 +1,244 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::GPIO;
+use drv_spi_api::Spi;
+use drv_stm32h7_eth as eth;
+use drv_stm32h7_gpio_api as gpio_api;
+use ksz8463::{Ksz8463, Register as KszRegister};
+use ringbuf::*;
+use userlib::{hl::sleep_for, task_slot};
+use vsc7448_pac::types::PhyRegisterAddress;
+use vsc85xx::{Phy, PhyRw, PhyVsc85xx, VscError};
+
+task_slot!(SPI, spi_driver);
+const KSZ8463_SPI_DEVICE: u8 = 0; // Based on app.toml ordering
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum Trace {
+    None,
+    Ksz8463Status { port: u8, status: u16 },
+    Vsc8552Status { port: u8, status: u16 },
+}
+ringbuf!(Trace, 16, Trace::None);
+
+////////////////////////////////////////////////////////////////////////////////
+
+pub struct Bsp {
+    ksz: Ksz8463,
+}
+
+impl Bsp {
+    pub fn new() -> Self {
+        let spi = Spi::from(SPI.get_task_id()).device(KSZ8463_SPI_DEVICE);
+        let ksz = Ksz8463::new(spi, gpio_api::Port::A.pin(0), false);
+
+        Self { ksz }
+    }
+
+    pub fn configure_ethernet_pins(&self) {
+        // This board's mapping:
+        //
+        // RMII REF CLK     PA1
+        // RMII RX DV       PA7
+        //
+        // RMII RXD0        PC4
+        // RMII RXD1        PC5
+        //
+        // RMII TX EN       PG11
+        // RMII TXD1        PG12
+        // RMII TXD0        PG13
+        //
+        // MDIO             PA2
+        //
+        // MDC              PC1
+        //
+        // (it's _almost_ identical to the STM32H7 Nucleo, except that
+        //  TXD1 is on a different pin)
+        //
+        //  The MDIO/MDC lines run at Speed::Low because otherwise the VSC8504
+        //  refuses to talk.
+        use gpio_api::*;
+        let gpio = Gpio::from(GPIO.get_task_id());
+        let eth_af = Alternate::AF11;
+
+        // RMII
+        gpio.configure(
+            Port::A,
+            (1 << 1) | (1 << 7),
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::VeryHigh,
```

Comment:
(FWIW, from the datasheet, VeryHigh may actually be higher than we need for 50MHz RMII, but it'll depend on capacitance.)